src/NexusMods.App.UI/Pages/NexusModsDataProvider.cs

```diff
@@ -56,58 +58,114 @@ public IObservable<IChangeSet<ILibraryItemModel, EntityId>> ObserveCollectionIte
                 CollectionDownloadsFilter.OnlyOptional => downloadEntity.IsOptional,
             }))
             .FilterImmutable(static downloadEntity => downloadEntity.IsCollectionDownloadNexusMods() || downloadEntity.IsCollectionDownloadExternal())
-            .Transform(ILibraryItemModel (downloadEntity) =>
+            .TransformOnObservable(IObservable<ILibraryItemModel> (downloadEntity) =>
             {
                 if (downloadEntity.TryGetAsCollectionDownloadNexusMods(out var nexusModsDownload))
                 {
-                    return ToLibraryItemModel(nexusModsDownload);
+                    return ToLibraryItemModelObservable(revisionMetadata, nexusModsDownload);
                 }
 
                 if (downloadEntity.TryGetAsCollectionDownloadExternal(out var externalDownload))
                 {
-                    return ToLibraryItemModel(externalDownload);
+                    return System.Reactive.Linq.Observable.Return(ToLibraryItemModel(externalDownload));
                 }
 
                 throw new UnreachableException();
             });
     }
 
-    private ILibraryItemModel ToLibraryItemModel(CollectionDownloadNexusMods.ReadOnly nexusModsDownload)
+    private IObservable<ILibraryItemModel> ToLibraryItemModelObservable(
+        CollectionRevisionMetadata.ReadOnly revisionMetadata,
+        CollectionDownloadNexusMods.ReadOnly nexusModsDownload)
     {
-        var isInLibraryObservable = CollectionDownloader.IsDownloadedObservable(_connection, nexusModsDownload)
-            .ToObservable()
-            .Prepend(nexusModsDownload, static download => CollectionDownloader.IsDownloaded(download, download.Db));
+        var isInLibraryObservable = CollectionDownloader
+            .IsDownloadedObservable(_connection, nexusModsDownload)
+            .Prepend(CollectionDownloader.IsDownloaded(nexusModsDownload, nexusModsDownload.Db));
+
+        // NOTE(erri120): different behavior depending on whether the download is optional or not.
+        // Optional downloads can be individually installed by the user, Required downloads should only
+        // show that they are installed, but not offer an install button.
+        var observable = nexusModsDownload.AsCollectionDownload().IsOptional ? isInLibraryObservable : isInLibraryObservable.SelectMany(IObservable<bool> (isInLibrary) =>
+        {
+            if (!isInLibrary) return System.Reactive.Linq.Observable.Return(false);
+            if (!CollectionDownloader.TryGetDownloadedItem(nexusModsDownload, _connection.Db, out var libraryItem)) throw new NotImplementedException();
+            return GetIsInstalledObservable(libraryItem.AsLibraryItem(), revisionMetadata);
+        }).DistinctUntilChanged();
```

Comment:
If you remove an optional download, it correctly returns ready for install.
If you remove some or all required mods you can't install them again at the moment, as the Install all button doesn't get enabled again.

What should be the expected behavior for required mods?

Reply:
Collections are supposed to be read-only, so you shouldn't be able to remove required mods.

Reply:
What is going to be the expected way to remove an installed collection from a loadout (without deleting it from the library as well)?

Reply:
Not sure, I haven't seen designs for that yet.

Reply:
Worth asking Design then, in any case not a problem for this PR